driver/ppm_fillers.c

```diff
@@ -1363,9 +1388,12 @@ static int parse_sockopt(struct event_filler_arguments *args, int level, int opt
 #ifdef SO_SNDTIMEO
 			case SO_SNDTIMEO:
 #endif
-				if (unlikely(ppm_copy_from_user(&u.tv, optval, sizeof(u.tv))))
+				ASSERT(optlen == sizeof(struct timeval));
```

Comment:
MAJOR: ASSERT doesn't really work in kernel space (we don't map it to BUG_ON/WARN_ON). Also, optlen comes from userspace, so it's not our bug if optlen != sizeof(struct timeval) but userspace shenanigans.